src/redux/books/books.js

```diff
@@ -1,33 +1,76 @@
+// import { createAsyncThunk } from '@reduxjs/toolkit';
+// import axios from 'axios';
+
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const GET_BOOKS = 'bookstore/books/GET_BOOKS';
 
-// reducer
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/z9HYwqfQehINXNpxFodv/books';
 
 // actions
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
-});
+export const addBook = (book) => async (dispatch) => {
+  await fetch(url, {
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then(() => dispatch({ type: ADD_BOOK, book }));
+};
+
+export const removeBook = (index) => async (dispatch) => {
+  await fetch(`${url}/${index}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then(() => dispatch({ type: REMOVE_BOOK, index }));
+};
+
+export const getBooks = () => async (dispatch) => {
+  await fetch(url)
+    .then((res) => res.json())
+    .then((book) => {
+      const booksArray = [];
+      Object.keys(book).forEach((key) => {
+        booksArray.push({
+          item_id: key,
+          author: book[key][0].author,
+          title: book[key][0].title,
+          category: book[key][0].category,
+        });
+      });
+
+      dispatch({ type: GET_BOOKS, booksArray });
+    });
+};
```

Comment:
- [ ] It is a project requirement to use `createAsyncThunk` in the project for creating thunk action creators. Kindly do that

Reply:
@GzimAsani Thanks for your comments. Actually, I was trying to implement `createAsyncThunk`, but it wasn't working. Therefore, I commented that out. Moreover, it isn't a requirement. Please check out the screenshot below. 
It says **Fetch OR Axios with createAsyncThunk**. We can use fetch as well.

Waiting for your reply.
![2022-07-15_14-23-22](https://user-images.githubusercontent.com/78845635/179195402-e699a2a8-3eb1-4cb8-ae84-dd9e72e51cf7.jpg)

Reply:
Hello @safar1212, I apologize I read it wrong I missed the `OR` part, kindly remove the commented out code and you should be good :+1: